lua/tardis/screens/sh_virtualconsole.lua

```diff
@@ -235,5 +272,150 @@ TARDIS:AddScreen("Controls", {menu=false}, function(self,ext,int,frame,screen)
+	power:SetControl("power")
+	power:SetPressedStateData(ext, "power-state")
+	layout:AddNewButton(power)
+
+	local repair = TardisScreenButton:new(frame)
+	repair:SetIsToggle(true)
+	repair:SetText("repair")
+	repair:SetControl("repair")
+	repair:SetPressedStateData(ext, "repair-primed")
+	layout:AddNewButton(repair)
+
+	local fastremat = TardisScreenButton:new(frame)
+	fastremat:SetIsToggle(true)
+	fastremat:SetText("fast_remat")
+	fastremat:SetControl("fastremat")
+	fastremat:SetPressedStateData(ext, "demat-fast")
+	layout:AddNewButton(fastremat)
+
+	local throttle = TardisScreenButton:new(frame)
+	throttle:SetIsToggle(true)
+	throttle:SetText("throttle")
+	throttle:SetControl("teleport")
+	throttle:SetPressedStateData(ext, "teleport", "vortex")
+	layout:AddNewButton(throttle)
+
+	local fastreturn = TardisScreenButton:new(frame)
+	fastreturn:SetIsToggle(false)
+	fastreturn:SetText("fast_return")
+	fastreturn:SetControl("fastreturn")
+	layout:AddNewButton(fastreturn)
+
+	local flight = TardisScreenButton:new(frame)
+	flight:SetIsToggle(true)
+	flight:SetText("flight")
+	flight:SetControl("flight")
+	flight:SetPressedStateData(ext, "flight")
+	layout:AddNewButton(flight)
+
+	local float = TardisScreenButton:new(frame)
+	float:SetIsToggle(true)
+	float:SetText("anti-gravs")
+	float:SetControl("float")
+	float:SetPressedStateData(ext, "float")
+	layout:AddNewButton(float)
+
+	local physlock = TardisScreenButton:new(frame)
+	physlock:SetIsToggle(true)
+	physlock:SetText("physlock")
+	physlock:SetControl("physbrake")
+	physlock:SetPressedStateData(ext, "physlock")
+	layout:AddNewButton(physlock)
+
+	local hads = TardisScreenButton:new(frame)
+	hads:SetIsToggle(true)
+	hads:SetText("hads")
+	hads:SetControl("hads")
+	hads:SetPressedStateData(ext, "hads")
+	layout:AddNewButton(hads)
+
+	local destination = TardisScreenButton:new(frame)
+	destination:SetIsToggle(false)
+	destination:SetText("manual_destination_select")
+	destination:SetControl("destination")
+	layout:AddNewButton(destination)
+
+	local flightcontrol = TardisScreenButton:new(frame)
+	flightcontrol:SetIsToggle(false)
+	flightcontrol:SetText("flight_control")
+	flightcontrol:SetControl("flightcontrol")
+	layout:AddNewButton(flightcontrol)
+
+	layout:DrawButtons()
+
+	layout.scroll_size = math.max(1, math.floor(layout:GetCols() / 2))
+
+	frame.Think = function()
+		screen.left_arrow:SetVisible(true)
+		screen.right_arrow:SetVisible(true)
+		screen.right_arrow.DoClick = function()
+			if layout:CanMoveLeft()
+				and not screen.left_arrow:IsPressed()
+				and not screen.right_arrow:IsPressed()
+			then
+				layout:ScrollButtons(-layout.scroll_size)
+			end
+		end
+		screen.left_arrow.DoClick = function()
+			if layout:CanMoveRight()
+				and not screen.right_arrow:IsPressed()
+				and not screen.left_arrow:IsPressed()
+			then
+				layout:ScrollButtons(layout.scroll_size)
+			end
+		end
+	end
+
+end
+
+TARDIS:AddScreen("Console", {menu=false}, function(self,ext,int,frame,screen)
```

Comment:
Should this be called Virtual Console?

Reply:
Well, main screen buttons are currently sorted by name, so it would always be the last, even at small screens.
No difference for Capaldi, but for legacy/2005 smaller screens you have to scroll each time to get to them.
I'm sure that the text versions wouldn't be used frequently or used at all.

Maybe I will added other order/sorting in the future updates, but for now it's better to leave it first than last

Reply:
Happy for this to be left for now, but I would like to see an ordering system and for this to be renamed back to Virtual Console, I will raise an issue for this post-merge and assign it to you

Reply:
Closing this conversation with issue moved to:
https://github.com/MattJeanes/TARDIS/issues/360#issuecomment-784152484